src/testcases/org/apache/poi/xssf/usermodel/helpers/XSSFColumnShifterTest.java

```diff
@@ -0,0 +1,384 @@
+package org.apache.poi.xssf.usermodel.helpers;
+
+import static org.apache.poi.POITestCase.skipTest;
+import static org.apache.poi.POITestCase.testPassesNow;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeTrue;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.poi.common.usermodel.HyperlinkType;
+import org.apache.poi.ss.ITestDataProvider;
+import org.apache.poi.ss.usermodel.*;
+import org.apache.poi.ss.util.CellAddress;
+import org.apache.poi.ss.util.CellRangeAddress;
+import org.apache.poi.ss.util.CellUtil;
+import org.apache.poi.xssf.XSSFITestDataProvider;
+import org.apache.poi.xssf.XSSFTestDataSamples;
+import org.apache.poi.xssf.usermodel.*;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XSSFColumnShifterTest {
+
+	//private static Logger log = LoggerFactory.getLogger(XSSFColumnShifterTest.class + "_T");
+	private XSSFSheet sheet1, sheet2;
+	private Workbook wb07;
+
+    protected final ITestDataProvider _testDataProvider;
+
+	public XSSFColumnShifterTest(){
+        _testDataProvider = XSSFITestDataProvider.instance; 
+    }
+
+	@Before
+	public void init() {
+		wb07 = new XSSFWorkbook();
+		sheet1 = (XSSFSheet) wb07.createSheet("sheet1");
+		XSSFRow row = sheet1.createRow(0);
+		row.createCell(0, CellType.NUMERIC).setCellValue(0);
+		row.createCell(1, CellType.NUMERIC).setCellValue(1);
+		XSSFCell c1 = row.createCell(2, CellType.NUMERIC);
+		c1.setCellValue(2);
+
+		row = sheet1.createRow(1);
+		row.createCell(0, CellType.NUMERIC).setCellValue(0.1);
+		row.createCell(1, CellType.NUMERIC).setCellValue(1.1);
+		row.createCell(2, CellType.NUMERIC).setCellValue(2.1);
+		row = sheet1.createRow(2);
+		row.createCell(0, CellType.NUMERIC).setCellValue(0.2);
+		row.createCell(1, CellType.NUMERIC).setCellValue(1.2);
+		row.createCell(2, CellType.NUMERIC).setCellValue(2.2);
+		row = sheet1.createRow(3);
+		row.createCell(0, CellType.FORMULA).setCellFormula("A2*B3");
+		row.createCell(1, CellType.NUMERIC).setCellValue(1.3);
+		row.createCell(2, CellType.FORMULA).setCellFormula("B1-B3");
+		row = sheet1.createRow(4);
+		row.createCell(0, CellType.FORMULA).setCellFormula("SUM(C1:C4)");
+		row.createCell(1, CellType.FORMULA).setCellFormula("SUM(A3:C3)");
+		row.createCell(2, CellType.FORMULA).setCellFormula("$C1+C$2");
+		row = sheet1.createRow(5);
+		row.createCell(1, CellType.NUMERIC).setCellValue(1.5);
+		/*
+		 * sheet2 = (XSSFSheet)wb07.createSheet("sheet2"); row =
```

Comment:
What is the plan for all the commented out test code?

Reply:
Now I have removed comment marks.
But I want to know is there a common method for displaying sheets in console in POI. If there is one, I'd like to use that instead of my writeSheetToLog() method. If there is not, would you like me to create it (it should be a bit more subtle than writeSheetToLog()) ?

Reply:
I don't know of a util like this - we can add one another time if is useful but for now, just keep the write method local to your test.